packages/settings-library/requirements/ci.txt

```diff
@@ -12,6 +12,7 @@
 
 # installs this repo's packages
 pytest-simcore @ ../pytest-simcore
+simcore-models-library @ ../models-library
```

Comment:
I also think this is wrong!

@pcrespov did we change our minds about these dependencies?

Reply:
I'm not a fan of shared libraries, they are **bad** especially in Microservices Architecture because the create dependencies and _tight coupling_.

The `settings-library` depends on `models-library` because we need a place where to put the `OsparcErrorMixin` since they changed the previous behavior of `PydanticErrorMixin`.

Classes and in general "support" stuff like this can be used everywhere and don't change due to new requirements.

IMO, we need a place - e.g. a `common-library` - in which we can place these _domain agnostic_ things and keep in `models-library`, as the name suggests, the _domain models_ .

Reply:
If the problem is having a "common-library" with more basic functionality. Got ahead. This has the advantage that we do a separation of domains but NOTE that at the end, when used on the service, we are installing ALL of them.

There is another limitation in our installation setup when it comes to dependencies between osparc packages if we compare with third-party packages. As example. If a library A depends on B, we just add it in the A's requirement. If B depends on C, you do NOT need to add C in the former requirements list.  When A, B and C happen to be osparc packages, we need to add A, B and also C in the requirements listings.

Perhaps it is time to review this approach. At the time I did not find a way to treat dependencies between osparc packages as any other requirement! 

@giancarloromeo I think this is easier explained ini person

Reply:
@giancarloromeo

I would also support this `common-library` where you put inside the `OsparcErrorMixin` and the remaining items that you require to share in this same manner.

If you guys have a better idea, I would still to it, but after this big PR that migrates to pydantic2. I'd like to not mix too many things as this is already pretty challenging.